src/core/exec-invoke.c

```diff
@@ -5506,10 +5588,27 @@ int exec_invoke(
                         command,
                         needs_sandboxing,
                         have_cap_sys_admin,
+                        bpffs_socket_fd,
                         exit_status);
         if (r < 0)
                 return r;
 
+        if (context->private_bpf != PRIVATE_BPF_NO) {
+                r = pidref_wait_for_terminate_and_check("(sd-bpffs)", &bpffs_pid, /* flags = */ 0);
+                TAKE_PIDREF(bpffs_pid);
+                if (r < 0) {
+                        *exit_status = EXIT_BPF;
+                        return r;
+                }
+                /* If something strange happened with the child, let's consider this fatal, too */
+                if (r != EXIT_SUCCESS) {
+                        *exit_status = EXIT_BPF;
+                        if (read(bpffs_errno_pipe, &r, sizeof(r)) == sizeof(r))
+                                return log_debug_errno(r, "bpffs helper exited with error: %m");
+                        return -EIO;
```

Comment:
Please debug log when we can't read the error as well. And use `errno` so we know which error `read()` failed with instead of `EIO`.